clippy_lints/src/trait_bounds.rs

```diff
@@ -40,11 +40,12 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TraitBounds {
             if let WherePredicate::BoundPredicate(ref p) = bound {
                 let h = hash(&p.bounded_ty);
                 if !set.insert(h) {
-                    span_lint(
+                    span_help_and_lint(
                         cx,
                         TYPE_REPETITION_IN_BOUNDS,
                         p.span,
-                        "This type has already been used as a bound predicate",
+                        "this type has already been used as a bound predicate",
+                        "consider combining the bounds: `T: Copy + Clone`",
```

Comment:
Hmm, I don't like that this help message is hardcoded. Do you think it would be possible to determine all bounds and then produce a help message with theses bounds?

I'm thinking of turning the `set` into a `map: HashMap<WhereBoundPredicate, Vec<GenericBounds>>` and then using 2 for-loops: in the first iteration you build the `map` and in the second iteration you lint on elements in that `map` where the `Vec<_>` contains more then one item and build the suggestion from this `Vec`.

Relevant doc: https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/struct.WhereBoundPredicate.html

Reply:
I thought that might the feedback, but it let me see I was using `span_help_and_lint` correctly. I'm currently updating rustc and rebasing so I can do the `ty` changes but I'll do this bit next